nautobot/core/views/generic.py

```diff
@@ -197,10 +200,39 @@ def get(self, request):
         content_type = ContentType.objects.get_for_model(model)
 
         filter_params = request.GET
+        user = request.user
         display_filter_params = []
         dynamic_filter_form = None
         filter_form = None
         hide_hierarchy_ui = False
+        clear_view = filter_params.get("clear_view", False)
+
+        # If the user clicks on the clear view button, we do not check for global or user defaults
+        if not clear_view:
+            # Check if there is a default for this view for this specific user
+            app_label, model_name = model._meta.label.split(".")
+            view_name = f"{app_label}:{model_name.lower()}_list"
+
+            if not isinstance(user, AnonymousUser):
+                user_default_saved_view_pk = user.config_data.get("saved_views", {}).get(view_name, None)
+                if user_default_saved_view_pk is not None and not request.GET.get("saved_view"):
+                    try:
+                        SavedView.objects.get(pk=user_default_saved_view_pk)
+                        sv_url = reverse("users:savedview", kwargs={"pk": user_default_saved_view_pk})
+                        return redirect(sv_url)
+                    except ObjectDoesNotExist:
+                        # Saved view was deleted
+                        pass
+
+            # Check if there is a global default for this view
+            global_saved_view = None
+            try:
+                global_saved_view = SavedView.objects.get(view=view_name, is_global_default=True)
+            except ObjectDoesNotExist:
+                pass
+
+            if global_saved_view is not None and not request.GET.get("saved_view"):
```

Comment:
If an explicit `saved_view` was specified, we should be able to skip all of the above lookups - no need to look for a user default view or a global default view...?

Same comment on the UIViewSet code.

Reply:
Sorry if I am not understanding this correctly. The second part of the if statement `not request.GET.get("saved_view")` is basically acting as you said skipping the check if there is a saved view specified.

Reply:
Sorry - my point was that it's a waste of time to look up the user default and global default saved views if we can know in advance that we don't need them. So something like:

```
if not request.GET.get("saved_view"):
    # look for user default view if any, and apply it if found
    # else look for global default view if any, and apply it if found
```

would be more efficient and possibly cleaner as well.